ESSArch_Core/test_util.py

```diff
@@ -46,3 +57,125 @@ def test_zero_returncode_with_file_created(self, mock_popen, mock_isfile):
+
+
+class GetFilesAndDirsTest(TestCase):
+
+    def test_get_files_and_dirs_none_existent_dir(self):
+        self.assertEqual(get_files_and_dirs("this_folder_or_path_should_not_exist"), [])
+
+
+class ParseContentRangeHeaderTest(TestCase):
+
+    def test_parse_content_range_header(self):
+        header = 'bytes 123-456/789'
+        ref_start = 123
+        ref_end = 456
+        ref_total = 789
+
+        (start, end, total) = parse_content_range_header(header)
+        self.assertEqual(start, ref_start)
+        self.assertEqual(end, ref_end)
+        self.assertEqual(total, ref_total)
+
+    def test_parse_content_range_header_with_bad_header(self):
+        header = "bad header"
+        from rest_framework.exceptions import ValidationError
+
+        with self.assertRaises(ValidationError) as context:
+            parse_content_range_header(header)
+
+        self.assertTrue('Invalid Content-Range header' in str(context.exception))
+
+
+class FlattenTest(TestCase):
+
+    def test_flatten_list_of_lists(self):
+        my_list = [
+            list(range(1, 10)),
+            list(range(10, 20)),
+        ]
+
+        result_list = list(range(1, 20))
+        self.assertEqual(result_list, flatten(my_list))
+
+
+class GetSchemasTest(TestCase):
+
+    def setUp(self):
+        self.datadir = tempfile.mkdtemp()
+        self.addCleanup(shutil.rmtree, self.datadir)
+
+    def get_simple_valid_xml(self):
+        tmp_file = tempfile.TemporaryFile(dir=self.datadir)
+        tmp_file.write(b'<?xml version="1.0" encoding="ISO-8859-1"?>')
+        tmp_file.write(b'<vaxml xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+                       b'xmlns="http://www.visualarkiv.se/vaxml/v6.1" '
+                       b'xsi:schemaLocation="http://www.visualarkiv.se/vaxml/v6.1 '
+                       b'http://www.visualarkiv.se/vaxml/v6.1/vaxml.xsd" created="2019-01-23T19:17:34" '
+                       b'generator="Visual Arkiv 6.2 version 6.2.0.12" countrycode="SE"></vaxml>')
+        tmp_file.seek(0)
+
+        return tmp_file
+
+    def get_bad_xml_file(self):
+        tmp_file = tempfile.TemporaryFile(dir=self.datadir)
+        tmp_file.write(b'Hello bad XML syntax!')
+        tmp_file.seek(0)
+
+        return tmp_file
+
+    def test_get_schemas_from_doc(self):
+        filename = self.get_simple_valid_xml()
+
+        parser = etree.XMLParser(remove_blank_text=True)
+        doc = etree.parse(filename, parser=parser)
+
+        schema = getSchemas(doc=doc)
+        self.assertTrue(type(schema) is etree.XMLSchema)
+
+    def test_get_schema_from_file(self):
+        schema = getSchemas(filename=self.get_simple_valid_xml())
+        self.assertTrue(type(schema) is etree.XMLSchema)
+
+    def test_get_schema_with_no_argument_should_throw_exception(self):
+        with self.assertRaises(AttributeError) as context:
+            getSchemas()
+
+        self.assertTrue("'NoneType' object has no attribute 'getroot'" in str(context.exception))
```

Comment:
Use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex to validate exception message